### PR TITLE
Adds outline and selective bloom 

### PR DESCRIPTION
**Pull request by gsimone**
- [x] Adds Outline effect
- [x] Adds SelectiveBloom effect
- [x] Adds example for both effects

### CODE REVIEW

File: src/effects/Outline.tsx [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
+    ...props
+}: OutlineProps,
   ref: Ref<OutlineEffect>
 ) {
   const { scene, camera } = useThree()
+
   const effect = useMemo(
-    () =>
-      new OutlineEffect(scene, camera, {
+    () =>{
```
Comment (stephencorwin) [inline src/effects/Outline.tsx:24]: It looks like maybe prettier failed on commit here. I'll submit a PR shortly to convert to a more similar setup to the v5 branch in r3f

File: src/effects/ColorAverage.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { ColorAverageEffect, BlendFunction } from 'postprocessing'
 import React, { Ref, ForwardRefExoticComponent, forwardRef, useMemo } from 'react'
 
 export type ColorAverageProps = Partial<{
-  blendFunction: number
+  blendFunction?: BlendFunction
```
Comment (talentlessguy) [inline src/effects/ColorAverage.tsx:5]: `Partial` type already makes properties optional. No need for `?` :D

ts doc: https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype

File: src/effects/Outline.tsx [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 
-type OutlineProps = ConstructorParameters<typeof OutlineEffect>[2]
+type ObjectRef = MutableRefObject<Object3D>
 
-export const Outline: ForwardRefExoticComponent<OutlineEffect> = forwardRef(function Outline(
-  props: OutlineProps,
+export type OutlineProps = ConstructorParameters<typeof OutlineEffect>[2] &
+  Partial<{
+    selection?: ObjectRef | ObjectRef[]
+    selectionLayer?: number
```
Comment (talentlessguy) [inline src/effects/Outline.tsx:11]: same here, `Partial<T>` u don't have to put `?`

File: src/effects/SelectiveBloom.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { EffectComposerContext } from '../EffectComposer'
+
+type ObjectRef = MutableRefObject<Object3D>
+
+export type SelectiveBloomProps = ConstructorParameters<typeof SelectiveBloomEffect>[2] &
+  Partial<{
+    lights?: ObjectRef[]
+    selection?: ObjectRef | ObjectRef[]
+    selectionLayer?: number
+  }>
```
Comment (talentlessguy) [inline src/effects/SelectiveBloom.tsx:13]: ....and same for here....

**Review by talentlessguy — CHANGES_REQUESTED** [review]
Pls remove `?` from partial cuz it already makes all properties optional

**Comment by talentlessguy** [comment]
I will add the example to the PR, to see it works as it should :D

**Comment by CharlieHess** [comment]
I synced this branch with `yarn add https://github.com/react-spring/react-postprocessing.git/#adds-select-effects` to test the outline effect and found selection not updating on mouse over. My legacy effects chain from three/examples works with the same code so something might be wrong here.

**Review by stephencorwin — APPROVED** [review]

**Pull request merged**